lib/matplotlib/cbook/__init__.py

```diff
@@ -2250,6 +2250,13 @@ def _unpack_to_numpy(x):
     if isinstance(x, np.ndarray):
         # If numpy, return directly
         return x
+    if hasattr(x, '__array__'):
+        xtmp = np.array(x)
```

Comment:
```suggestion
        xtmp = np.asarray(x)
```

Which will avoid a copy if possible.

Reply:
Maybe best to limit the fix to the place touched by #22141? (`_check_1d`). Really, that should perhaps use `_unpack_to_numpy()`, but perhaps leave that for a bigger update...